### DISCUSSION TITLE
Apply colormap based on min/max in current bbox

### DISCUSSION THREAD

**Discussion by nicokant** [developmentseed/titiler#772] [Q&A]
Hi,
I'm trying to understand if it would be possible using this library to apply a colormap to a COG normalizing based on the min/max of the bounding box instead of the whole raster.
In case it would be possible, could you suggest what are the options/strategies to achive that?
Thank you!

**Comment by vincentsarago** [comment]
Hi @nicokant
I think you could totally do this using custom algorithm https://developmentseed.org/titiler/advanced/Algorithms/

```python
from titiler.core.algorithm import BaseAlgorithm
from rio_tiler.models import ImageData

class UpdateStats(BaseAlgorithm):

    def __call__(self, img: ImageData) -> ImageData:

        stats = []
        for ix in range(img.count):
            stat_min, stat_max = img.array[ix].min(), img.array[ix].max()
            stats.append((stat_min, stat_max))

        img.dataset_statistics = stats 
        return img
```
